airflow/api_fastapi/core_api/routes/public/config.py

```diff
@@ -0,0 +1,122 @@
+from fastapi.responses import Response
+
+from airflow.api_fastapi.common.router import AirflowRouter
+from airflow.api_fastapi.core_api.datamodels.config import (
+    Config,
+    ConfigOption,
+    ConfigSection,
+)
+from airflow.api_fastapi.core_api.openapi.exceptions import create_openapi_http_exception_doc
+from airflow.configuration import conf
+
+config_router = AirflowRouter(tags=["Config"], prefix="/config")
+
+
+@config_router.get(
+    "/",
+    responses=create_openapi_http_exception_doc(
+        [status.HTTP_401_UNAUTHORIZED, status.HTTP_403_FORBIDDEN, status.HTTP_404_NOT_FOUND]
+    ),
+)
+def get_config(
+    section: str | None = None,
+    content_type: Literal["application/json", "text/plain"] = Header(...),
+) -> Response:
+    if conf.get("webserver", "expose_config").lower() == "non-sensitive-only":
+        expose_config = True
+        display_sensitive = False
+    else:
+        expose_config = conf.getboolean("webserver", "expose_config")
+        display_sensitive = True
+
+    if not expose_config:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="Your Airflow administrator chose not to expose the configuration, most likely for security reasons.",
+        )
+    if section and not conf.has_section(section):
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"Section {section} not found.",
+        )
+    conf_dict = conf.as_dict(display_source=False, display_sensitive=display_sensitive)
+
+    if section:
+        conf_section_value = conf_dict[section]
+        conf_dict.clear()
+        conf_dict[section] = conf_section_value
+
+    config = Config(
+        sections=[
+            ConfigSection(
+                name=section, options=[ConfigOption(key=key, value=value) for key, value in options.items()]
+            )
+            for section, options in conf_dict.items()
+        ]
+    )
+
+    return Response(
+        content=config.model_dump_json() if content_type == "application/json" else config.text_format,
+        headers={"Content-Type": content_type},
+    )
+
+
+@config_router.get(
+    "/section/{section}/option/{option}",
+    responses=create_openapi_http_exception_doc(
+        [status.HTTP_401_UNAUTHORIZED, status.HTTP_403_FORBIDDEN, status.HTTP_404_NOT_FOUND]
+    ),
+)
+def get_config_value(
+    section: str,
+    option: str,
+    content_type: Literal["application/json", "text/plain"] = Header(...),
+) -> Response:
+    if conf.get("webserver", "expose_config").lower() == "non-sensitive-only":
+        expose_config = True
+    else:
+        expose_config = conf.getboolean("webserver", "expose_config")
+
+    if not expose_config:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="Your Airflow administrator chose not to expose the configuration, most likely for security reasons.",
+        )
```

Comment:
These lines are redundant to lines 46-62 - can you push this to a common function and use this from both methods?

Reply:
Sure, just resolved. Thanks for the reminder!